### ISSUE TITLE
PoB Performance Optimization

### ISSUE THREAD

**Issue by dsnvwlmnt** [PathOfBuildingCommunity/PathOfBuilding#5985]
### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### Is your feature request related to a problem?

There are a couple areas within the client that I've noticed perform far less than ideally:
- Loading a build for the first time
- Skills: Loading the support gem list when you click a new Gem name slot

These are the two big ones I can remember offhand.

### Describe the solution you'd like

Faster performance. There is surely room for improvement here.

### Additional context

For context I'm running Win10 on a gaming desktop Ryzen7 1700 8-core, 16gb ram, GTX 1060. Most everything else in PoB (and other software) is very snappy.

**Comment by QuickStick123** [comment]
We already cache all the mods and gem sorting (but we have to rebuild a gem slot because item data is relevant). I don't believe it will be we can make it much faster without somehow magically making all the code run faster. There may be potential but PoB is written in lua which isn't the most performant and would require significant reworks that I highly doubt would happen anytime soon.

If you believe it is so easy to optimise PoB sure you can have a go at it and open a PR accomplishing this?

**Comment by dsnvwlmnt** [comment]
Fair enough, I assumed not a lot of optimization had been done, given its performance inadequacies.

Posting feedback does not require someone to be inclined to do it. I'm merely pointing out a flaw with the program.
